### DISCUSSION TITLE
Question -- upgrade path

### DISCUSSION THREAD

**Discussion by rchavez-neu** [samvera/serverless-iiif#127] [Q&A]
a genera question:  is there a preferred/recommended upgrade path when deploying the serverless-iiif via the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)?

Specifically, can an existing deploy be updated in place when using the AWS SAP, or does this method of deploy always deploy a new instance of serverless-iiif?

Whatever the case, it would be good to add a bit about upgrades and/or best practices for handling them in the documentation.  Would be happy to help do that if needed.

**Comment by mbklein** [comment]
AWS SAR does a really good job upgrading an app stack. You just redeploy the app as if you're starting over, and give it the same application name. More detail is available in AWS' [Updating Applications](https://docs.aws.amazon.com/serverlessrepo/latest/devguide/serverlessrepo-how-to-consume-new-version.html) docs.

**Comment by mbklein** [comment]
The previous answer assumes you're trying to upgrade the same “flavor” of serverless-iiif (serverless-iiif <= 3.x /  serverless-iiif-standalone or serverless-iiif-cloudfront 4.x) that you installed the first time. Switching from one to the other probably requires a full redeploy. Some documentation on that would be great and I'd welcome a pull request.

We're also looking to add a dedicated github.io site with better multi-page documentation at some point. The README is getting a little unwieldy and (IMO) intimidating.

**Comment by rchavez-neu** [comment]
Thank you @mbklein, I appreciate the info.  I will give this a try in the coming weeks and put in a PR with some updated doc for the README once I get clear of some ongoing projects.

**Comment by mbklein** [comment]
@rchavez-neu We're actually very close to a new release, with a single-stream deploy and much more extensive documentation, so I would suggest holding off for a few days. The initial upgrade might be a little tricky, but it's going to make things _much_ easier in the long run.

**Comment by mbklein** [comment]
@rchavez-neu The [new version](https://serverlessrepo.aws.amazon.com/applications/us-east-1/625046682746/serverless-iiif) was released a few weeks ago. Please take a look and see if you have any further questions.
